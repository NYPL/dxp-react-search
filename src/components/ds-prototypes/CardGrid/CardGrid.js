import React, { Fragment } from 'react';
// Components
import { 
  Heading,
  List
} from '@nypl/design-system-react-components';

function CardGrid(props) {
  const { title, description, children } = props;

  return (
    <div className="card-grid">
      <Heading
        id={title}
        level={2}
        text={title}
      />
      {description &&
        <p>{description}</p>
      }
      <List
        className="card-grid__list"
        modifiers={[
          'no-list-styling'
        ]}
        type="ul"
      >
        {children}
      </List>
    </div>
  );
};

export default CardGrid;