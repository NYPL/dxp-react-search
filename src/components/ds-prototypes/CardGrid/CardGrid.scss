.card-grid {
  .card-grid__list {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
    grid-gap: 1.25rem;

    .heading {
      padding-top: var(--space-xs);

      a {
        color: black;
      }
    }

    .image {
      width: 100%;
      height: auto;
    }
  }
}