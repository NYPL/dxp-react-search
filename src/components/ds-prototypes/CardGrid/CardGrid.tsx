import React, { Fragment } from "react";

interface CardGridProps {
  //
  templateColumns: string;
  //
  gap: string;
  /** The children of the card grid */
  children: any;
}

function CardGrid({ templateColumns, gap, children }: CardGridProps) {
  return (
    <ul
      style={{
        listStyleType: "none",
        padding: 0,
        margin: 0,
        display: "grid",
        gridTemplateColumns: templateColumns,
        gridGap: gap,
      }}
    >
      {children}
    </ul>
  );
}

export default CardGrid;
