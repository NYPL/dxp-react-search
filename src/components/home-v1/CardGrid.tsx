import React from "react";
import { Grid, Heading, Box } from "@chakra-ui/react";
import ComponentWrapper from "./ComponentWrapper";

interface CardGridProps {
  title: string;
  items: CardGridItem[];
}

interface CardGridItem {
  id: string;
  title: string;
  description?: string;
  image?: string;
}

function CardGrid({ title, items }: CardGridProps) {
  return (
    <ComponentWrapper
      title={title}
      textColor="red.200"
      borderColor="red"
      paddingTop={true}
    >
      <Grid
        as="ul"
        listStyleType="none"
        templateColumns={{
          base: "1fr",
          md: "repeat(auto-fit, minmax(267px, 1fr))",
        }}
        w="full"
        gap={6}
      >
        {items.map((item: CardGridItem) => (
          <li key={item.id}>
            <Box display={"flex"} flexDirection={{ base: "row", md: "column" }}>
              <Box w={{ base: "30vw", md: "100%" }}>
                <img width="100%" height="auto" src={item.image} />
              </Box>
              <Heading
                as="h3"
                size="md"
                w={{ base: "60vw", md: "auto" }}
                fontWeight="bold"
                fontFamily="Kievit-Book"
                my={2}
                mx={{ base: "5vw", md: "0px" }}
              >
                {item.title}
              </Heading>
            </Box>
          </li>
        ))}
      </Grid>
    </ComponentWrapper>
  );
}

export default CardGrid;
