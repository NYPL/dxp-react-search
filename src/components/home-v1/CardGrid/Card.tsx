import React from "react";
import { useStyleConfig } from "@chakra-ui/system";
// Components
import { Heading, Text, Grid, GridItem, Box } from "@chakra-ui/react";
import Image, { ImageType } from "./../../shared/Image";
import HomePageLink, { AnalyticsEventActions } from "../HomePageLink";
export interface CardItem {
  id: string;
  title: string;
  description?: string;
  image: ImageType;
  url?: string;
}

interface CardProps {
  item: CardItem;
  variant?: "blog-card" | "updates-card";
  size?: any;
  analyticsEventActions: AnalyticsEventActions;
  gaEventActionName: string;
}

function Card({
  item,
  variant,
  size = "md",
  analyticsEventActions,
  gaEventActionName,
}: CardProps) {
  // Get Card theme styles
  const styles = useStyleConfig("Card", { variant, size });
  // Generate describedBy string (used by Blog Card)
  const describedByIdsArray = [];
  const omitItems = ["id", "title", "image", "url", "__typename"];
  for (const propName in item) {
    if (!omitItems.includes(propName)) {
      describedByIdsArray.push(`card-${propName}-${item.id}`);
    }
  }
  const describedByIdsString = describedByIdsArray.join(" ");

  return (
    <Grid
      id={`card-${item.id}`}
      templateRows={{ base: "1fr", md: "min-content" }}
      templateColumns={{ base: "5fr 9fr", md: "1fr" }}
      sx={styles}
    >
      <GridItem className="textBox">
        <Heading
          as="h3"
          {...(describedByIdsString && {
            "aria-describedby": describedByIdsString,
          })}
        >
          <HomePageLink
            href={item.url}
            analyticsEventActions={analyticsEventActions}
            gaEventActionName={gaEventActionName}
          >
            {item.title}
          </HomePageLink>
        </Heading>
        <Box className="details">
          {item.description && (
            <Text id={`card-description-${item.id}`}>{item.description}</Text>
          )}
        </Box>
      </GridItem>
      <GridItem id={`card-image-${item.id}`} colStart={1} rowStart={1}>
        <HomePageLink
          id={`card-image-link-${item.id}`}
          href={item.url}
          aria-label={`${item.title}-image`}
          analyticsEventActions={analyticsEventActions}
          gaEventActionName={gaEventActionName}
          tabIndex={-1}
        >
          {/* 
            // @QUESTION should role="presentation" be used instead of alt="" source: https://www.digitala11y.com/presentation-role/
            role="presentation"
          */}
          <Image
            id={`card-image-${item.image.id}`}
            alt={item.image.alt}
            uri={item.image.uri}
            useTransformation={true}
            transformations={item.image.transformations}
            transformationLabel={"2_1_960"}
            layout="responsive"
            width={900}
            height={450}
            quality={90}
          />
        </HomePageLink>
      </GridItem>
    </Grid>
  );
}

export default Card;
