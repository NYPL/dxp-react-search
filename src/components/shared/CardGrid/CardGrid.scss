.card-grid {
  .card-grid__list {
    display: flex;
    flex-flow: row wrap;
    padding: .75rem 0;

    .heading {
      padding-top: 1rem;
    }

    .image {
      width: 384px;
      height: 256px;
    }
  }
}