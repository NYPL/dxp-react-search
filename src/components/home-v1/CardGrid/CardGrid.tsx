import React from "react";
import { useStyleConfig } from "@chakra-ui/system";
// Components
import { Grid, GridItem } from "@chakra-ui/react";
import ComponentWrapper from "../ComponentWrapper";
import Card, { CardItem } from "./Card";

interface CardGridProps {
  title: string;
  link: string;
  items: CardItem[];
  hoverStyle?: boolean;
  variant?: "row-grid" | "column-grid" | "updates-grid";
  //props below go inform styling of LinkCard
  cardVariant?: "event-spotlight" | "event-card" | "blog-card" | "updates-card";
  size?: string | Record<string, string>;
}

function CardGrid({
  title,
  link,
  items,
  hoverStyle = false,
  variant,
  cardVariant,
  size,
}: CardGridProps) {
  const styles = useStyleConfig("CardGrid", { variant });
  return (
    <ComponentWrapper
      title={title}
      link={link}
      hoverStyle={hoverStyle}
      //last three could become passed through props
      textColor="red.200"
      borderColor="red.200"
      paddingTop={true}
    >
      <Grid as="ul" sx={styles}>
        {items &&
          items.map((item) => (
            <GridItem as="li" listStyleType="none">
              <Card
                item={item}
                size={size || { base: "sm", md: "lg" }}
                variant={cardVariant}
              />
            </GridItem>
          ))}
      </Grid>
    </ComponentWrapper>
  );
}

export default CardGrid;
