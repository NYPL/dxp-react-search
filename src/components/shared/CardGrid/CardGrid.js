import React, { Fragment } from 'react';
// Components
import { 
  Heading,
  Image,
  Link,
  List,
  SkeletonLoader 
} from '@nypl/design-system-react-components';
import Card from './../Card';

function CardGrid(props) {
  const { title, items } = props;
  // items have id, title, description, imageUrl props

  return (
    <div className="card-grid">
      <Heading
        id={title}
        level={2}
        text={title}
      />
      <List
        className="card-grid__list"
        modifiers={[
          'no-list-styling'
        ]}
        type="ul"
      >
        {items.map((item) => {
          return (
            <li 
              key={item.id} 
              className="card-grid__list-item"
              style={{
                'margin-bottom': '2rem',
                'flex': '1 0 calc(33% - 2rem)',
                'margin-right': '2rem',
                'max-width': '384px'
              }}
            >
              <Card
                name={item.name}
                imageUrl={item.imageUrl}
                description={item.description} 
              />
            </li>
          )
        })}
      </List>
    </div>
  );
};

export default CardGrid;
