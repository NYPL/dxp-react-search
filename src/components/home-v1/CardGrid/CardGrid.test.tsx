import * as React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import "@testing-library/jest-dom/extend-expect";
expect.extend(toHaveNoViolations);
import renderer from "react-test-renderer";
// Component
import CardGrid from "./CardGrid";

const items = [
  {
    id: "test-id-1",
    title: "Test",
    image: {
      id: "test-id-1",
      alt: "image-1-alt-text",
      layout: "responsive",
      width: 400,
      height: 200,
      quality: 90,
      uri: "https://placeimg.com/400/200/arch",
      useTransformation: false,
      transformationLabel: "2_1_960",
    },
    url: "https://www.nypl.org/",
  },
  {
    id: "test-id-2",
    title: "Test",
    image: {
      id: "test-id-2",
      alt: "image-2-alt-text",
      layout: "responsive",
      width: 400,
      height: 200,
      quality: 90,
      uri: "https://placeimg.com/400/200/arch",
      useTransformation: false,
      transformationLabel: "2_1_960",
    },
    url: "https://www.nypl.org/",
  },
  {
    id: "test-id-3",
    title: "Test",
    image: {
      id: "test-id-3",
      alt: "image-3-alt-text",
      layout: "responsive",
      width: 400,
      height: 200,
      quality: 90,
      uri: "https://placeimg.com/400/200/arch",
      useTransformation: false,
      transformationLabel: "2_1_960",
    },
    url: "https://www.nypl.org/",
  },
  {
    id: "test-id-4",
    title: "Test",
    image: {
      id: "test-id-4",
      alt: "image-4-alt-text",
      layout: "responsive",
      width: 400,
      height: 200,
      quality: 90,
      uri: "https://placeimg.com/400/200/arch",
      useTransformation: false,
      transformationLabel: "2_1_960",
    },
    url: "https://www.nypl.org/",
  },
];

describe("CardGrid tests", () => {
  it("should pass axe accessibility test with children components", async () => {
    const { container } = render(
      <CardGrid title="Test title" link="https://nypl.com" items={items} />
    );
    await waitFor(async () =>
      expect(await axe(container)).toHaveNoViolations()
    );
  });
  it("should render list of items passed to the component", async () => {
    render(
      <CardGrid title="Test title" link="https://nypl.com" items={items} />
    );
    await waitFor(() =>
      expect(screen.getAllByRole("listitem")).toHaveLength(4)
    );
  });
  it("should render the UI snapshot correctly", async () => {
    const cardGrid = renderer
      .create(
        <CardGrid title="Test title" link="https://nypl.com" items={items} />
      )
      .toJSON();
    await waitFor(() => expect(cardGrid).toMatchSnapshot());
  });
});
