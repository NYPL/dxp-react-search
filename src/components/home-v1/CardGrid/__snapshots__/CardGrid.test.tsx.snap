// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests should render the UI snapshot correctly 1`] = `
<div
  className="css-pgu20y"
  id="card-test-id-1"
>
  <div
    className="textBox css-0"
  >
    <h3
      aria-describedby="card-location-test-id-1"
      className="chakra-heading css-0"
    >
      <a
        className="css-0"
        href="https://www.nypl.org/"
        onClick={[Function]}
      >
        Test
      </a>
    </h3>
    <div
      className="details css-0"
    />
  </div>
  <div
    className="css-ys3583"
    id="card-image-test-id-1"
  >
    <a
      aria-label="Test-image"
      className="css-0"
      href="https://www.nypl.org/"
      id="card-image-link-test-id-1"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="scout-nextjs-image"
        id="nextjsImage-card-image-test-id-1"
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "opacity": 1,
                "padding": 0,
                "paddingTop": "50%",
                "width": "initial",
              }
            }
          />
          <img
            alt="image-1-alt-text"
            data-nimg="responsive"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="image-1-alt-text"
              data-nimg="responsive"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=3840&q=90"
              srcSet="/_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=640&q=90 640w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=750&q=90 750w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=828&q=90 828w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1080&q=90 1080w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1200&q=90 1200w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1920&q=90 1920w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=2048&q=90 2048w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=3840&q=90 3840w"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </div>
    </a>
  </div>
</div>
`;

exports[`CardGrid tests should render the UI snapshot correctly 1`] = `
<div
  className="css-15j2i3p"
>
  <div
    className="css-1ngwrio"
  >
    <div
      className="css-hze00u"
    >
      <div
        className="css-1uxn9gx"
      >
        <h2
          className="chakra-heading css-18pkilg"
          id="component-wrapper-heading-card-grid-test-id"
        >
          <a
            className="css-0"
            href="https://nypl.com"
            id="component-wrapper-heading-link-card-grid-test-id"
            onClick={[Function]}
          >
            Test title
          </a>
        </h2>
      </div>
    </div>
    <div
      className="css-1y2jlch"
    >
      <ul
        className="css-lgj0h8"
      >
        <li
          className="css-155za0w"
          id="card-grid-item-test-id-1"
        >
          <div
            className="css-pgu20y"
            id="card-test-id-1"
          >
            <div
              className="textBox css-0"
            >
              <h3
                className="chakra-heading css-0"
              >
                <a
                  className="css-0"
                  href="https://www.nypl.org/"
                  onClick={[Function]}
                >
                  Test
                </a>
              </h3>
              <div
                className="details css-0"
              />
            </div>
            <div
              className="css-ys3583"
              id="card-image-test-id-1"
            >
              <a
                aria-label="Test-image"
                className="css-0"
                href="https://www.nypl.org/"
                id="card-image-link-test-id-1"
                onClick={[Function]}
                tabIndex={-1}
              >
                <div
                  className="scout-nextjs-image"
                  id="nextjsImage-card-image-test-id-1"
                >
                  <span
                    style={
                      Object {
                        "background": "none",
                        "border": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": "initial",
                        "margin": 0,
                        "opacity": 1,
                        "overflow": "hidden",
                        "padding": 0,
                        "position": "relative",
                        "width": "initial",
                      }
                    }
                  >
                    <span
                      style={
                        Object {
                          "background": "none",
                          "border": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": "initial",
                          "margin": 0,
                          "opacity": 1,
                          "padding": 0,
                          "paddingTop": "50%",
                          "width": "initial",
                        }
                      }
                    />
                    <img
                      alt="image-1-alt-text"
                      data-nimg="responsive"
                      decoding="async"
                      onError={[Function]}
                      onLoad={[Function]}
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style={
                        Object {
                          "border": "none",
                          "bottom": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": 0,
                          "left": 0,
                          "margin": "auto",
                          "maxHeight": "100%",
                          "maxWidth": "100%",
                          "minHeight": "100%",
                          "minWidth": "100%",
                          "objectFit": undefined,
                          "objectPosition": undefined,
                          "padding": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "width": 0,
                        }
                      }
                    />
                    <noscript>
                      <img
                        alt="image-1-alt-text"
                        data-nimg="responsive"
                        decoding="async"
                        loading="lazy"
                        sizes="100vw"
                        src="/_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=3840&q=90"
                        srcSet="/_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=640&q=90 640w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=750&q=90 750w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=828&q=90 828w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1080&q=90 1080w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1200&q=90 1200w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1920&q=90 1920w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=2048&q=90 2048w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=3840&q=90 3840w"
                        style={
                          Object {
                            "border": "none",
                            "bottom": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": 0,
                            "left": 0,
                            "margin": "auto",
                            "maxHeight": "100%",
                            "maxWidth": "100%",
                            "minHeight": "100%",
                            "minWidth": "100%",
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "padding": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": 0,
                          }
                        }
                      />
                    </noscript>
                  </span>
                </div>
              </a>
            </div>
          </div>
        </li>
        <li
          className="css-155za0w"
          id="card-grid-item-test-id-2"
        >
          <div
            className="css-pgu20y"
            id="card-test-id-2"
          >
            <div
              className="textBox css-0"
            >
              <h3
                className="chakra-heading css-0"
              >
                <a
                  className="css-0"
                  href="https://www.nypl.org/"
                  onClick={[Function]}
                >
                  Test
                </a>
              </h3>
              <div
                className="details css-0"
              />
            </div>
            <div
              className="css-ys3583"
              id="card-image-test-id-2"
            >
              <a
                aria-label="Test-image"
                className="css-0"
                href="https://www.nypl.org/"
                id="card-image-link-test-id-2"
                onClick={[Function]}
                tabIndex={-1}
              >
                <div
                  className="scout-nextjs-image"
                  id="nextjsImage-card-image-test-id-2"
                >
                  <span
                    style={
                      Object {
                        "background": "none",
                        "border": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": "initial",
                        "margin": 0,
                        "opacity": 1,
                        "overflow": "hidden",
                        "padding": 0,
                        "position": "relative",
                        "width": "initial",
                      }
                    }
                  >
                    <span
                      style={
                        Object {
                          "background": "none",
                          "border": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": "initial",
                          "margin": 0,
                          "opacity": 1,
                          "padding": 0,
                          "paddingTop": "50%",
                          "width": "initial",
                        }
                      }
                    />
                    <img
                      alt="image-2-alt-text"
                      data-nimg="responsive"
                      decoding="async"
                      onError={[Function]}
                      onLoad={[Function]}
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style={
                        Object {
                          "border": "none",
                          "bottom": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": 0,
                          "left": 0,
                          "margin": "auto",
                          "maxHeight": "100%",
                          "maxWidth": "100%",
                          "minHeight": "100%",
                          "minWidth": "100%",
                          "objectFit": undefined,
                          "objectPosition": undefined,
                          "padding": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "width": 0,
                        }
                      }
                    />
                    <noscript>
                      <img
                        alt="image-2-alt-text"
                        data-nimg="responsive"
                        decoding="async"
                        loading="lazy"
                        sizes="100vw"
                        src="/_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=3840&q=90"
                        srcSet="/_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=640&q=90 640w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=750&q=90 750w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=828&q=90 828w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1080&q=90 1080w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1200&q=90 1200w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1920&q=90 1920w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=2048&q=90 2048w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=3840&q=90 3840w"
                        style={
                          Object {
                            "border": "none",
                            "bottom": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": 0,
                            "left": 0,
                            "margin": "auto",
                            "maxHeight": "100%",
                            "maxWidth": "100%",
                            "minHeight": "100%",
                            "minWidth": "100%",
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "padding": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": 0,
                          }
                        }
                      />
                    </noscript>
                  </span>
                </div>
              </a>
            </div>
          </div>
        </li>
        <li
          className="css-155za0w"
          id="card-grid-item-test-id-3"
        >
          <div
            className="css-pgu20y"
            id="card-test-id-3"
          >
            <div
              className="textBox css-0"
            >
              <h3
                className="chakra-heading css-0"
              >
                <a
                  className="css-0"
                  href="https://www.nypl.org/"
                  onClick={[Function]}
                >
                  Test
                </a>
              </h3>
              <div
                className="details css-0"
              />
            </div>
            <div
              className="css-ys3583"
              id="card-image-test-id-3"
            >
              <a
                aria-label="Test-image"
                className="css-0"
                href="https://www.nypl.org/"
                id="card-image-link-test-id-3"
                onClick={[Function]}
                tabIndex={-1}
              >
                <div
                  className="scout-nextjs-image"
                  id="nextjsImage-card-image-test-id-3"
                >
                  <span
                    style={
                      Object {
                        "background": "none",
                        "border": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": "initial",
                        "margin": 0,
                        "opacity": 1,
                        "overflow": "hidden",
                        "padding": 0,
                        "position": "relative",
                        "width": "initial",
                      }
                    }
                  >
                    <span
                      style={
                        Object {
                          "background": "none",
                          "border": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": "initial",
                          "margin": 0,
                          "opacity": 1,
                          "padding": 0,
                          "paddingTop": "50%",
                          "width": "initial",
                        }
                      }
                    />
                    <img
                      alt="image-3-alt-text"
                      data-nimg="responsive"
                      decoding="async"
                      onError={[Function]}
                      onLoad={[Function]}
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style={
                        Object {
                          "border": "none",
                          "bottom": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": 0,
                          "left": 0,
                          "margin": "auto",
                          "maxHeight": "100%",
                          "maxWidth": "100%",
                          "minHeight": "100%",
                          "minWidth": "100%",
                          "objectFit": undefined,
                          "objectPosition": undefined,
                          "padding": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "width": 0,
                        }
                      }
                    />
                    <noscript>
                      <img
                        alt="image-3-alt-text"
                        data-nimg="responsive"
                        decoding="async"
                        loading="lazy"
                        sizes="100vw"
                        src="/_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=3840&q=90"
                        srcSet="/_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=640&q=90 640w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=750&q=90 750w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=828&q=90 828w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1080&q=90 1080w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1200&q=90 1200w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1920&q=90 1920w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=2048&q=90 2048w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=3840&q=90 3840w"
                        style={
                          Object {
                            "border": "none",
                            "bottom": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": 0,
                            "left": 0,
                            "margin": "auto",
                            "maxHeight": "100%",
                            "maxWidth": "100%",
                            "minHeight": "100%",
                            "minWidth": "100%",
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "padding": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": 0,
                          }
                        }
                      />
                    </noscript>
                  </span>
                </div>
              </a>
            </div>
          </div>
        </li>
        <li
          className="css-155za0w"
          id="card-grid-item-test-id-4"
        >
          <div
            className="css-pgu20y"
            id="card-test-id-4"
          >
            <div
              className="textBox css-0"
            >
              <h3
                className="chakra-heading css-0"
              >
                <a
                  className="css-0"
                  href="https://www.nypl.org/"
                  onClick={[Function]}
                >
                  Test
                </a>
              </h3>
              <div
                className="details css-0"
              />
            </div>
            <div
              className="css-ys3583"
              id="card-image-test-id-4"
            >
              <a
                aria-label="Test-image"
                className="css-0"
                href="https://www.nypl.org/"
                id="card-image-link-test-id-4"
                onClick={[Function]}
                tabIndex={-1}
              >
                <div
                  className="scout-nextjs-image"
                  id="nextjsImage-card-image-test-id-4"
                >
                  <span
                    style={
                      Object {
                        "background": "none",
                        "border": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": "initial",
                        "margin": 0,
                        "opacity": 1,
                        "overflow": "hidden",
                        "padding": 0,
                        "position": "relative",
                        "width": "initial",
                      }
                    }
                  >
                    <span
                      style={
                        Object {
                          "background": "none",
                          "border": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": "initial",
                          "margin": 0,
                          "opacity": 1,
                          "padding": 0,
                          "paddingTop": "50%",
                          "width": "initial",
                        }
                      }
                    />
                    <img
                      alt="image-4-alt-text"
                      data-nimg="responsive"
                      decoding="async"
                      onError={[Function]}
                      onLoad={[Function]}
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style={
                        Object {
                          "border": "none",
                          "bottom": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": 0,
                          "left": 0,
                          "margin": "auto",
                          "maxHeight": "100%",
                          "maxWidth": "100%",
                          "minHeight": "100%",
                          "minWidth": "100%",
                          "objectFit": undefined,
                          "objectPosition": undefined,
                          "padding": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "width": 0,
                        }
                      }
                    />
                    <noscript>
                      <img
                        alt="image-4-alt-text"
                        data-nimg="responsive"
                        decoding="async"
                        loading="lazy"
                        sizes="100vw"
                        src="/_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=3840&q=90"
                        srcSet="/_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=640&q=90 640w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=750&q=90 750w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=828&q=90 828w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1080&q=90 1080w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1200&q=90 1200w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=1920&q=90 1920w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=2048&q=90 2048w, /_next/image?url=https%3A%2F%2Fplaceimg.com%2F400%2F200%2Farch&w=3840&q=90 3840w"
                        style={
                          Object {
                            "border": "none",
                            "bottom": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": 0,
                            "left": 0,
                            "margin": "auto",
                            "maxHeight": "100%",
                            "maxWidth": "100%",
                            "minHeight": "100%",
                            "minWidth": "100%",
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "padding": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": 0,
                          }
                        }
                      />
                    </noscript>
                  </span>
                </div>
              </a>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="css-1ktje31"
    >
      <div
        className="css-i2lecr"
      />
    </div>
  </div>
</div>
`;
